### DISCUSSION TITLE
Beforeunload Event Not Triggering in Chrome in Next.js App

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#72803] [Help]
### Summary

We’re experiencing an issue in our Next.js application where the default browser alert for unsaved changes does not trigger in Chrome upon browser/tab close or page reload. This behavior works as expected in Firefox and Edge, where the alert dialog is shown. Our application uses file-based routing, and we rely on the beforeunload event to prompt users about unsaved changes.

`"use client";
import { useState, useEffect } from 'react';

export default function Tab1() {
  const [inputVal, setInputVal] = useState('');

  useEffect(() => {
    const checkUnsavedChanges = (e: BeforeUnloadEvent) => {
      if (inputVal !== '') {
        e.preventDefault();
        e.returnValue = true; // Use this cautiously: it's needed but deprecated
      }
    };

    window.addEventListener('beforeunload', checkUnsavedChanges);

    return () => {
      window.removeEventListener('beforeunload', checkUnsavedChanges);
    };
  }, [inputVal]);

  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputVal(event.target.value);
  };

  return (
    <div className="App">
      <input 
        type="text" 
        value={inputVal} 
        onChange={handleInputChange} 
        style={{ border: "1px solid red" }}
      />
    </div>
  );
}
`

### Additional information

```js
When I create a new Next.js project, an alert is shown in Chrome, but it's not working in our source. I don't know how to debug or find the solution. Please help me with this.
```

**Comment by icyJoseph** [comment]
So it works if you start a new project, but not in your production codebase?

Add a debugger statement in your event listener, make sure that's running. 

What Chrome version are you testing this on?

**Reply by ghost** [reply]
Yes, it works when we create a new Next.js project, but it doesn't work in the production codebase. The debugger statement is not triggered in the beforeunload event listener in Edge either, but the browser alert does appear in edge.

Chrome version: 131.0.6778.70 (Official Build) (64-bit).

**Reply by icyJoseph** [reply]
Is there any user interaction on this page?

**Reply by ghost** [reply]
Yes, I have an input element in it. I type something, focus out, and then reload

**Comment by Tergeltengis** [comment]
Is there any solution?

**Comment by rubanraj54** [comment]
Facing same issue, it works on Firefox, but it doesn't work in chrome for some reason.
